assignments/arrays.js

```diff
@@ -65,32 +65,94 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
 console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
-
+function searchForCarInfo(a_id, list) {
+	var index = null;
+
+	// for loop to conduct search, for edge case that the car id's are unordered; 
+	for(i = 0; i < list.length; i++) {
+		if (list[i].id == a_id) {
+      
+			index = i;
+			break;	
+    	}
+	}
+  
+  return "Car " + list[index].id + " is a " + list[index].car_make + " " + list[index].car_model + " " + list[index].car_year + ".";
+}
+
+console.log(searchForCarInfo(33,inventory));
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
 let lastCar = 0;
-console.log();
+
+console.log(inventory[lastCar]);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+function retrieveAllCarModels(list) {
+	var CarModels = [];
+	for(i = 0; i < list.length; i++) {
+		if (!CarModels.includes(list[i].car_model)) {
+			CarModels.push(list[i].car_model);
+		}
+	}
+
+  CarModels.sort();
+	return CarModels; 
+}
+
+console.log(retrieveAllCarModels(inventory));
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
+function retrieveAllCarYears(list) {
+	var CarYears = [];
+	for(i = 0; i < list.length; i++) {
+		if (!CarYears.includes(list[i].car_year)) {
+			CarYears.push(list[i].car_year);
+		}
+	}
+  
+  CarYears.sort()
+	return CarYears; 
+}
+
+console.log(retrieveAllCarYears(inventory));
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+function checkNumCarsOlderThan(list,year) {
+	var number = 0; 
+		for(i = 0; i < list.length; i++) {
```

Comment:
Be careful here, this may work, but you should declare the variable `i` and `let` is the best choice.  var does some very weird stuff especially when it does this thing called hoisting which can lead to some funky behavior.